### PR TITLE
Implement Basin Hopping

### PR DESCRIPTION
**Pull request by Janix4000**

### CODE REVIEW

**Comment by Janix4000** [comment]
[CC-rule implementation: Simulated annealing](https://trello.com/c/X1vCepyE/38-cc-rule-implementation-simulated-annealing)

File: votepy/algorithms/basinhopping.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+import scipy
+import numpy as np
+import random
+
+from typing import Callable, Iterable
+
+
+@algo('basinhopping')
+# .
```
Comment (msluszniak) [inline votepy/algorithms/basinhopping.py]:
```suggestion
```

**Review by Wolokin — APPROVED** [review]
very nice

File: votepy/algorithms/basinhopping.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    def _solve(self, voting: OrdinalElection, size_of_committee: int):
+        if size_of_committee == voting.ballot_size:
+            return list(range(size_of_committee))
+
+        chosen = np.array(self.x0) if self.x0 is not None else np.arange(size_of_committee, dtype=int)
+        rest = np.array(list(set(range(voting.ballot_size)) - set(chosen)))
+
+        print(chosen, rest)
+
+        # chosen = np.arange(size_of_committee, dtype=int)
```
Comment (msluszniak) [inline votepy/algorithms/basinhopping.py]:
```suggestion
```

File: votepy/algorithms/basinhopping.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        # rest = np.arange(size_of_committee, voting.ballot_size, dtype=int)
```
Comment (msluszniak) [inline votepy/algorithms/basinhopping.py]:
```suggestion
```

**Review by msluszniak — APPROVED** [review]

**Pull request merged**